lib/internal/url.js

```diff
@@ -0,0 +1,328 @@
+'use strict';
+
+function getPunycode() {
```

Comment:
note: this is in anticipation of my other PR that switches to ICU's punycode impl landing. If/when that moves forward, we'll be able to take advantage of some other performance improvements from that.